### ISSUE TITLE
Making nano chatgpt

### ISSUE THREAD

**Issue by nebyu08** [karpathy/nanoGPT#92]

**Comment by Spiritdude** [comment]
Even though it expands beyond the scope of nanoGPT, a nanoChatGPT also came to my mind: replace web-site/blog search engine with nanoChatGPT answering questions based on information of a limited yet strong focused textset or factset.

I suggest we keep this thread/issue open so people can comment with links of "nano" ChatGPT like projects.

**Comment by zhzLuke96** [comment]
chatgpt training process is publicly available, but the results are highly dependent on the fine-tuned datasets that openai, which is hard to do

I think we can try webgpt, which is a more engineered solution

https://arxiv.org/abs/2112.09332
https://openai.com/blog/webgpt/

----
of course these are far beyond the scope of nanoGpt or Makemore courses
but, exercises after class are really fun

**Comment by Spiritdude** [comment]
Another effort at https://open-assistent.io, code at https://github.com/LAION-AI/Open-Assistant

**Comment by zhzLuke96** [comment]
I saw a project about chatgpt and felt compelled to share
https://github.com/hpcaitech/ColossalAI

They have made a very magical optimization of the training process. It is said that chatgpt(small) can be fine-tuned on a single card, and there is a complete training process code

**Comment by sanjeevanahilan** [comment]
I've started building a nanoChatGPT as a fork of Karpathy's chatGPT. I also introduce a new idea for training by backpropagating through the reward function using the Gumbel-Softmax trick rather than policy gradient (PPO).

It's still very crude though and far from being useful at scale. Sharing here in case anyone wants to try:

https://github.com/sanjeevanahilan/nanoChatGPT

**Comment by Spiritdude** [comment]
https://github.com/togethercomputer/OpenChatKit

**Comment by PiotrNawrot** [comment]
We have released [nanoT5](https://github.com/PiotrNawrot/nanoT5) for pre-training and evaluating T5-style (Encoder-Decoder) models. You can use it to pre-train your own model in one day on a single GPU :).

**Comment by VatsaDev** [comment]
Huh, There's actually an issue with my project ideas name on it. I have a NanoChatGPT [here](https://github.com/VatsaDev/nanoChatGPT) It has Chat functionality, human, bot, and endOfText tokens, along with a conversational dataset and more. I'm working on crude RLHF-like functionality, and would love contributors
